presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/ActualProperties.java

```diff
@@ -199,6 +205,18 @@ public static Builder builderFrom(ActualProperties properties)
         return new Builder(properties.global, properties.localProperties, properties.constants);
     }
 
+    private Map<Symbol, NullableValue> translateConstants(Function<Symbol, Optional<Symbol>> translator)
+    {
+        Map<Symbol, NullableValue> translatedConstants = new HashMap<>();
+        for (Map.Entry<Symbol, NullableValue> entry : constants.entrySet()) {
```

Comment:
Using `forEach` for maps is generally better than `entrySet` since you can name the key/value without having to unpack the entry. Also, `isPresent` can often be replaced with `ifPresent`:
```java
constants.forEach((symbol, value) -> translator.apply(symbol)
        .ifPresent(key -> translatedConstants.put(key, entry.getValue())));

Reply:
This code was part of the original `translate` function. I just moved it to a separate function since the same logic would be used in the newly introduced `translate` function as well. I can change it if you like.